htdocs/api/class/api.class.php

```diff
@@ -56,7 +56,7 @@ function __construct($db, $cachedir='', $refreshCache=false)
         $this->db = $db;
         $production_mode = ( empty($conf->global->API_PRODUCTION_MODE) ? false : true );
         $this->r = new Restler($production_mode, $refreshCache);
-
+	$this->r->setBaseUrls(DOL_MAIN_URL_ROOT);
```

Comment:
DOL_MAIN_URL_ROOT is "same domain than the current view page". But if you access using localhost, you will get localhost.

If you do:
$urlwithouturlroot=preg_replace('/'.preg_quote(DOL_URL_ROOT,'/').'$/i','',trim($dolibarr_main_url_root));
$urlwithroot=$urlwithouturlroot.DOL_URL_ROOT;		// This is to use external domain name found into config file

Then   $urlwithroot  is   the domain defined to be the external URL, always.


It seems we can set serveral url, so it may be better to add both of them

$this->r->setBaseUrls(DOL_MAIN_URL_ROOT, $urlwithroot);

Does this work for you as expected ?

Reply:
Yes. it does the trick.